packages/Parallel/Command/WorkerCommandLineFactory.php

```diff
@@ -93,7 +93,7 @@ public function create(
 
         if ($input->hasOption(Option::CONFIG)) {
             $workerCommandArray[] = '--config';
-            $workerCommandArray[] = $input->getOption(Option::CONFIG);
+            $workerCommandArray[] = escapeshellarg($input->getOption(Option::CONFIG));
```

Comment:
Could you add not to "why"? This might be copied later, so we knout about this issue.

Also what OS is it related, for future debugging :)

Reply:
on parallel, the command generated is:

```
```

Reply:
Note added: https://github.com/rectorphp/rector-src/pull/2421/commits/52d6f969908ba0080825aa5c2a21d6715ec972fe

On parallel, the command generated is:

```bash
'/opt/homebrew/Cellar/php/8.1.6/bin/php' '/Users/samsonasik/www/rector-src/e2e/../bin/rector' worker --dry-run --autoload-file '/Users/samsonasik/www/rector-src/e2e/../vendor/autoload.php' --clear-cache --port 51751 --identifier 'd449b0syyq' --output-format 'json' --no-ansi --config /Users/samsonasik/www/rector-src/e2e/parallel with space/rector.php
```

without the `escapeshellarg`, the `--config` as above, got the following:

```bash
--config /path/e2e/parallel with space/rector.php
```

which cause error:

```
Run php ../e2eTestRunner.php
{"fatal_errors":["File \"/home/runner/work/rector-src/rector-src/e2e/parallel\" was not found"]}
```

like in https://github.com/rectorphp/rector-src/runs/6723373768?check_suite_focus=true#step:8:9

the escaped result is:

```
--config '/path/e2e/parallel with space/rector.php'
```

tested in macOs (local) and Ubuntu (Github action).

Reply:
Thank you 😊